### ISSUE TITLE
Unique fields are required

### ISSUE THREAD

**Issue by AminZibayi** [strapi/strapi#3868]
**Describe the bug**
In fact, the unique fields are required even if you don't set the `required` property to true.

**Steps to reproduce the behavior**
1. Create a new Content-Type
2. Add a unique field and don't mark the required checkbox
5. Try to create at least two records of the created Content-Type without setting a value for the unique field
6. Encounter a 500 internal server error

**Expected behavior**
Since the field is not required we should be able to create records with no value specified for the unique field.

**System**
- Node.js version: v10.16.0
- NPM version: 6.9.0
- Strapi version: 3.0.0-beta.9
- Database: MongoDB
- Operating system: Ubuntu

**Additional context**
https://dba.stackexchange.com/questions/211522/mongodb-show-duplicate-key-is-null

**Comment by lauriejim** [comment]
Hello! Thank you for this report, appreciate.
This should be possible to have multiple null as a value for a unique field.
I will appreciate if you are able to investigate on it.

Check out the contributing guide to get started: https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md

The code should be in strapi-hook-mongoose.

**Comment by abdonrd** [comment]
I want to add more info about this!

I need a `Company` model with a `website` property. And as I expect that my companies never repeat the website, I set it as `"unique": true` and `"required": false`.

The problem is when I have two companies that don't have a `website`. Either the field is not in the database, or is like `null` or empty string (`""`). It complains that there is duplicity:

>E11000 duplicate key error collection: myapp.company index: website_1 dup key: { : "" }

Or:

>E11000 duplicate key error collection: myapp.company index: website_1 dup key: { : null }

As I set `"required": false`, I expect to allow this kind of "empty duplicity".

Maybe we should use [partial indexes](https://docs.mongodb.com/manual/core/index-partial/)?
